tests/scripts/checks.sh

```diff
@@ -2,35 +2,32 @@
 
 check_pod_ready() {
 	local pod_label=$1
-	local current_time=0
-	while :; do
-		echo "Checking $pod_label pod"
-		kubectl get pods -lapp=$pod_label -n ${TEST_NAMESPACE}
+	local pod_status_time_out=$2
+	
+	echo "Checking $pod_label pod"
+	
+	kubectl get pods -lapp=$pod_label -n ${TEST_NAMESPACE}
 
-		echo "Checking $pod_label pod readiness"
-		is_pod_ready=$(kubectl get pods -lapp=$pod_label -n ${TEST_NAMESPACE} -ojsonpath='{range .items[*]}{.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}' 2>/dev/null || echo "terminated")
+	echo "Checking $pod_label pod readiness"
 
-		if [ "${is_pod_ready}" = "True" ]; then
-			# Check if the pod is not in terminating state
-			is_pod_terminating=$(kubectl get pods -lapp=$pod_label -n ${TEST_NAMESPACE} -o jsonpath='{.items[0].metadata.deletionGracePeriodSeconds}' 2>/dev/null || echo "terminated")
-			if [ "${is_pod_terminating}" != "" ]; then
-				echo "pod $pod_label is in terminating state..."
-			else
-				echo "Pod $pod_label is ready"
-				break;
-			fi
-		fi
+	if kubectl wait -n ${TEST_NAMESPACE} --for=condition=Ready pod -l app=$pod_label --timeout ${pod_status_time_out}; then
+		return 0	
+	else
+		# print status of pod
+		kubectl get pods -n ${TEST_NAMESPACE}
+	fi
 
-		if [[ "${current_time}" -gt $((60 * 45)) ]]; then
-			echo "timeout reached"
-			exit 1;
-		fi
+	return 1
+}
 
-		# Echo useful information on stdout
-		kubectl get pods -n ${TEST_NAMESPACE}
+cleanup_pod(){
```

Comment:
Instead of implementing this function, create a new script file named `uninstall-operator.sh` that uninstalls the operator. Uninstalling the chart will remove all of the resources in the namespace, so you just need to run:

```
helm uninstall $release_name --namespace $namespace
```

Reply:
uninstall-operator.sh  created . 
helm uninstall $release_name --namespace $namespace was already there at the end of the function . 
_Uninstalling Helm itself will not remove DaemonSets or Secrets_